backend/internal/application/service/applicationservice.go

```diff
@@ -50,39 +50,18 @@ type ApplicationServiceInterface interface {
 
 // ApplicationService is the default implementation of the ApplicationServiceInterface.
 type ApplicationService struct {
-	AppStore    store.ApplicationStoreInterface
+	AppStore    store.CachedBackedApplicationStoreInterface
```

Comment:
Rather defining new `CachedBackedApplicationStoreInterface`, shall we use `ApplicationStoreInterface` making `NewCachedBackedApplicationStore` is a impl of `ApplicationStoreInterface`?

Reply:
Updated the method signature of `ApplicationStoreInterface` with d57a9fda417941b32a9b424c07cf8308b8179731